cmd/job-trigger-controller-manager/main.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/bombsimon/logrusr"
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/logrusutil"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrlruntimelog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	prpqv1 "github.com/openshift/ci-tools/pkg/api/pullrequestpayloadqualification/v1"
+	"github.com/openshift/ci-tools/pkg/controller/prpqr_reconciler"
+)
+
+const (
+	appCIContextName = "app.ci"
+)
+
+type options struct {
+	kubernetesOptions flagutil.KubernetesOptions
+
+	namespace string
+	dryRun    bool
+}
+
+func gatherOptions() (*options, error) {
+	o := &options{kubernetesOptions: flagutil.KubernetesOptions{NOInClusterConfigDefault: true}}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether to run the controller-manager with dry-run")
+	fs.StringVar(&o.namespace, "namespace", "", "In which namespace the operation will take place")
+	o.kubernetesOptions.AddFlags(fs)
+
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return o, fmt.Errorf("failed to parse flags: %w", err)
+	}
+	return o, nil
+}
+
+func (o *options) validate() error {
+	var errs []error
+
+	if o.namespace == "" {
+		errs = append(errs, fmt.Errorf("--namespace must be specified"))
+	}
+
+	if err := o.kubernetesOptions.Validate(false); err != nil {
+		errs = append(errs, err)
+	}
+	return utilerrors.NewAggregate(errs)
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	controllerruntime.SetLogger(logrusr.NewLogger(logrus.StandardLogger()))
+
+	o, err := gatherOptions()
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to get options")
+	}
+
+	if err := o.validate(); err != nil {
+		logrus.WithError(err).Fatal("Failed to validate options")
+	}
+
+	ctx := controllerruntime.SetupSignalHandler()
+	ctx, cancel := context.WithCancel(ctx)
+
+	kubeconfigChangedCallBack := func() {
+		logrus.Info("Kubeconfig changed, exiting to get restarted by Kubelet and pick up the changes")
+		cancel()
+	}
+
+	kubeconfigs, err := o.kubernetesOptions.LoadClusterConfigs(kubeconfigChangedCallBack)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to load kubeconfigs")
+	}
```

Comment:
I'd expect code like we do in other places:

https://github.com/openshift/ci-tools/blob/c3f8a3cdfc64de1120d0c56b8badd690e8c9a6eb/cmd/dptp-controller-manager/main.go#L247-L257